internal/paascharm/gogen.go

```diff
@@ -0,0 +1,81 @@
+package paascharm
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"path/filepath"
+	"text/template"
+)
+
+// CreateGoStructs reads the charmcraft.yaml file from the charmcraftFile location
+// and outputs the Go file with the proper structs in the outputFile location,
+// using as package name packageName. It will override the outputFile file if
+// it exists and create all the parent directories if missing.
+func CreateGoStructs(charmcraftFile string, packageName string, outputFile string) (err error) {
```

Comment:
Nit: style suggestion (compact format for args of the same type; it's also not typical to use named return values unless they're required for `defer` or useful to disambiguate):

```suggestion
func CreateGoStructs(charmcraftFile, packageName, outputFile string) error {
```

Reply:
thanks for the tip! done